mycore-base/src/main/resources/xsl/functions/derivate.xsl

```diff
@@ -0,0 +1,36 @@
+<?xml version="1.0"?>
+<xsl:stylesheet version="3.0"
+                xmlns:xsl="http://www.w3.org/1999/XSL/Transform"
+                xmlns:fn="http://www.w3.org/2005/xpath-functions"
+                xmlns:xs="http://www.w3.org/2001/XMLSchema"
+                xmlns:mcrderivate="http://www.mycore.de/xslt/derivate"
+                exclude-result-prefixes="fn xs">
+
+    <xsl:function name="mcrderivate:getMainFile" as="xs:string">
+        <xsl:param name="derivID" as="xs:string"/>
+        <xsl:variable name="derivate" select="document(concat('mcrobject:', $derivID))" />
+        <xsl:variable name="mainFile" select="$derivate/mycorederivate/derivate/internals/internal/@maindoc" />
+        <xsl:value-of select="$mainFile" />
+    </xsl:function>
+
+    <xsl:function name="mcrderivate:getContentType" as="xs:string">
```

Comment:
maybe better `mcrderivate:getFileContentType`?

Reply:
renamed to get-file-content-type